File: typescript/src/shared/ConsoleHelper.ts
/**
 * Shared Infrastructure — "Không-thời gian" mà modules sống trên.
 * Console output helpers, không chứa business logic.
 */
export function printHeader(title: string): void {
  const line = '═'.repeat(60);
  console.log();
  console.log(line);
  console.log(`  🌌 ${title}`);
  console.log(line);
}

export function printResult(module: string, command: string, args: string[], result: string): void {
  console.log(`  ▸ ${module} ${command} ${args.join(' ')}`);
  console.log(`    → ${result}`);
}

export function printBenchmark(label: string, iterations: number, elapsedMs: number): void {
  const opsPerSec = Math.round(iterations / (elapsedMs / 1000));
  console.log(`  ⚡ ${label}: ${iterations.toLocaleString()} ops in ${elapsedMs.toFixed(1)}ms (${opsPerSec.toLocaleString()} ops/sec)`);
}

File: python/shared/console_helper.py
"""
Shared Infrastructure — "Không-thời gian" mà modules sống trên.
Console output helpers, không chứa business logic.
"""
import sys

# Fix unicode output on Windows (cp1252 → utf-8)
sys.stdout.reconfigure(encoding="utf-8")


def print_header(title: str) -> None:
    line = "═" * 60
    print()
    print(line)
    print(f"  🌌 {title}")
    print(line)


def print_result(module: str, command: str, args: list[str], result: str) -> None:
    print(f"  ▸ {module} {command} {' '.join(args)}")
    print(f"    → {result}")


def print_benchmark(label: str, iterations: int, elapsed_ms: float) -> None:
    ops_per_sec = int(iterations / (elapsed_ms / 1000))
    print(f"  ⚡ {label}: {iterations:,} ops in {elapsed_ms:.1f}ms ({ops_per_sec:,} ops/sec)")
